### DISCUSSION TITLE
SPI with extension A0/DC

### DISCUSSION THREAD

**Discussion by ErikJensen1** [olikraus/u8g2#2352] [Ideas]
Many controllers have this extra pin A0/DC and it is not a problem for the parallel port solution except you need an extra output pin. If you use at serial solution like SPI this extra pin is difficult to implement and time consuming for the cpu. 
Could it be an idea to put data in a buffer with 9 bits and in the spi interrupt routine set/reset the A0/DC output pin and then send the 8 bit SPI data? Or are there any other smart solutions to solve this? Buy the display with another controller that avoid the A0/DC is of course a solution. 
Best regards
Erik

**Comment by olikraus** [comment]
Your idea with the 9 bit is exactly the 3-wire SPI solution available with many displays. Maybe your display already has this mode.
U8g2 also supports this mode. The mentioned DC / A0 pin then is obsolete.

**Reply by ErikJensen1** [reply]
I am afraid not. If I read the datasheet for controller ST7567, I cannot see this mode. But could be I am wrong?
Your need a display with ST7920 or others controllers to avoid A0/DC.
I have the display working now with this setup
u8g2_Setup_st7567_jlx12864_f(&u8g2_1, U8G2_R0, u8x8_byte_4wire_hw_spi, u8x8_stm32_gpio_and_delay);
and the DC msg is defined. I did not try without defining the DC msg. Do you think this will be possible?

**Reply by olikraus** [reply]
At least here https://www.buydisplay.com/download/ic/ST7567S.pdf this mode is mentioned on page 19.
